nightfall-optimist/src/services/transaction-sign-send.mjs

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-param-reassign */
+
+import config from 'config';
+import { web3 } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import { estimateGas, estimateGasPrice } from '@polygon-nightfall/common-files/utils/gas.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { Mutex } from 'async-mutex';
+
+const { GAS, GAS_MULTIPLIER, GAS_ESTIMATE_ENDPOINT, GAS_PRICE, GAS_PRICE_MULTIPLIER } = config;
+
+const nonceMutex = new Mutex();
+
+const isWeb3Listening = async () => {
+  try {
+    const isListening = await web3.eth.net.isListening();
+    return isListening;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+};
+
+export async function getAddressNonce(ethAddress) {
+  try {
+    const nonce = await web3.eth.getTransactionCount(ethAddress);
```

Comment:
You can return directly.

Reply:
Removed, see comment re `isWeb3Listening`